### PR TITLE
Lewd viro

### PR DESCRIPTION
**Pull request by ktccd**
:cl: ktccd
add: Added new citadel-only symptoms to viruses.
/:cl:

[why]: # 
Because we got these citadel-only things which could be used by virology. Mostly came to think about it when I figured if there was some way to get airborne aphrodisiac, when the word "airborne" made me think of virology.
# Ty to Snugglesnek for giving the idea of a virus that messes with genitalia. Sadly, the way we handle genitalia is very snow-flakey and not generic for all genitals, so yeah...

### CODE REVIEW

**Comment by ktccd** [comment]
Yeah, please **do not merge** yet, I still got one more commit coming up for this one, the PR is mostly open so I can get feedback in the meantime.

**Comment by Poojawa** [comment]
Probably pre-package these into a virus outbreak too, make it pick which to grow based on gender and already established organs maybe.

**Comment by ktccd** [comment]
So you mean make it so the viral outbreak can result in these?
Also, it doesn't care about gender, because genitals do not care about gender. It only checks if you have them, and if so it makes them grow a size.

**Comment by TalkingCactus** [comment]
I always wanted virus outbreak to be a bit more dangerous. Have it be a pandemic scenario where the virus mutates rapidly 'n stuff. Anyways, most transformations of genitals will be placed in procs when the system is more complete. Probably a good idea to hold off on this until then.

File: code/citadel/cit_virology.dm
Comment (TalkingCactus) [inline code/citadel/cit_virology.dm]: Useless comment, defines save on memory and are easy to maintain for constants that are referenced a lot.

File: code/citadel/cit_virology.dm
Comment (TalkingCactus) [inline code/citadel/cit_virology.dm]: `O.length = Clamp(O.length++, COCK_SIZE_MIN, COCK_SIZE_MAX)`
or
```
O.length++
if(O.length > COCK_SIZE_MAX) O.length = COCK_SIZE_MAX
```

File: code/citadel/cit_virology.dm
Comment (TalkingCactus) [inline code/citadel/cit_virology.dm]: `if(ishuman(A.affected_mob))`

File: code/citadel/cit_virology.dm
Comment (TalkingCactus) [inline code/citadel/cit_virology.dm]: Move this below the `..()` and change it to 
```
if(!A.affected_mob) return
```
Though I don't think this proc would even call if there wasn't an affected mob.

File: code/citadel/cit_virology.dm
Comment (TalkingCactus) [inline code/citadel/cit_virology.dm]: See Extendong comment; Same thing but inverse.

**Review by TalkingCactus — CHANGES_REQUESTED** [review]

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: yeah, I think I forgot to remove the comment after finding the constant. When I wrote the comment, I had just put "20" there XD.

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: so clamp forces it to stay between the boundaries?

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: oh, yeah. might be weird to recast non-humans to humans. heh...

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: I think my idea... was something like, M would be null if A wasn't human. But I was just bad at how recasting types worked. I think maybe the result of recasting A as human, if it wasn't human, would probably not be null after all.

File: code/citadel/cit_virology.dm
Comment (TalkingCactus) [reply code/citadel/cit_virology.dm]: Yes

File: code/citadel/cit_virology.dm
Comment (TalkingCactus) [reply code/citadel/cit_virology.dm]: Monkeys can carry viruses as well. Xenos might be able to as well, haven't tested that in a while.

Comment (ktccd) [reply code/citadel/cit_virology.dm]: so there's no point in doing if(M) then.
I guess it makes sense, none of the other viruses seem to do that.

Comment (TalkingCactus) [reply code/citadel/cit_virology.dm]: Mhmm

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: ✅

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: ✅

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: ✅

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: ✅

File: code/citadel/cit_virology.dm
Comment (ktccd) [reply code/citadel/cit_virology.dm]: ✅

**Comment by ktccd** [comment]
oh goodness...

**Comment by ktccd** [comment]
Ok, so, comparing the breasts_onmob.dmi with my commited one, I can't really spot why there's a conflict...
I renamed the sizes to use numbers instead of cupsizes, that's pretty much it...

**Comment by TalkingCactus** [comment]
https://github.com/Citadel-Station-13/Citadel-Station-13/commits/master/code/citadel/icons/breasts_onmob.dmi

**Comment by ktccd** [comment]
Wait, what did the other PR change with lighting?
There... isn't lighting in that file? O.o

**Comment by TalkingCactus** [comment]
I'm showing you where the changes occurred since you couldn't find it.

**Comment by ktccd** [comment]
Yeah, but... the file looks the same? Doesn't it?
I tried comparing with the file I took from master...

EDIT: I guess to be extra-sure, I'll just download the master version and rename the icon states.

**Comment by TalkingCactus** [comment]
They're changed, trust me. Look at the PR description for the most recent commit.

**Comment by ktccd** [comment]
The PR doesn't say what the exact pixel fix is, so I can't manually compare the files.

EDIT:
As it is now, I have downloaded the .dmi file from the master-branch and renamed the icon_states so they work with the new size system. After doing this change, git does not see any change when comparing the binary file from my local files with my commit here, so I assume the binary is identical now (Meaning the current commited version should be the same as the master-branch version + my name change).

**Comment by TalkingCactus** [comment]
Why do you need to compare them? Just take then newest one and rename them.

**Comment by ktccd** [comment]
I got no idea what is wrong with the .dme file though... I wasn't sure I changed that? Except maybe add in my new file to the list of files to compile?

**Comment by ktccd** [comment]
I did. But you know how git won't commit a file if it hasn't been changed? That's the issue I'm having.
Downloading the master-file and renaming the icon_state names results in a file that github can't see a difference in...

EDIT: i.e the file I've already committed to this PR is the master-file version with names changed, but it still conflicts because I made this PR a while ago and github can't compare binaries it seems.

**Comment by TalkingCactus** [comment]
Your only salvation may be a new PR, though I'm fairly certain there is a way to circumvent Git hating .dmi changes. As for the .dme, you just need to merge with the latest changes from the master and compile.

**Comment by ktccd** [comment]
Well, whoever handles merging can resolve conflicts I'm sure. I know there's a way to handle binary file conflicts, by simply saying which of the 2 to use over the other.
Also, the .dme... I mean... Won't merging master into this PR fuck up the list of commits? Like, won't it contain every commit merged to master for the last week?

**Comment by ktccd** [comment]
I mean, it's really freaking messy... but I could maybe just remove my .dme and .dmi changes, commit with master-versions of them and send a new PR with only those changed? It feels really messy though >_<.

**Comment by ktccd** [comment]
How can the .dme have conflicts when I took the file from master? What.....

File: code/citadel/organs/genitals.dm
Comment (TalkingCactus) [inline code/citadel/organs/genitals.dm]: These both should be combined into `AdjustSize(amt)` so it can accept an argument for changing the size. Look at the arousal setting procs for clues for achieving this. This proc would do `organ.size = (organ.size + amt)` and clamp it within the bounds set through the defines in `cit_defines.dm`. Doing this allows you to do increase and decreases in the same proc.

After that, add another proc called `SetSize(size)` which can be used to force a size change to a specific point.

**Review by TalkingCactus — CHANGES_REQUESTED** [review]

**Comment by ktccd** [comment]
Trying out this conflict solution. Praying to God it works....
I just added my virology file to the master-branch version of the .dme

**Comment by ktccd** [comment]
Since this compiled now, I'm just putting out a reminder that this is only up here for review.
*No merging* until... IDK. I need to make a 2nd PR with the breasts_onmob.dmi file to go with this, but I don't think I can MAKE the PR until this merges... and I don't want this to merge without it... since that'd make boobs not work without the file... :S

**Comment by ktccd** [comment]
WHAT. WHY? WHEN? HOW? It passed already!

**Comment by ktccd** [comment]
Resolved conflicts. I'm assuming whoever renamed breast shapes to "Pair" instead of "pair" is correct here and going with their changes.

**Comment by ktccd** [comment]
Alright, all conflicts resolved and the other PR is open. pinging @TalkingCactus to review this agin?

**Comment by ktccd** [comment]
Closing it 'cause even if it compiles, I can't actually test if it works after all the conflicts. That, and there's that thing with the .dmi file which kept giving me headaches.

**Pull request closed without merging**
